support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -181,52 +277,71 @@ function userInBreakpoint(audience: Audience): boolean {
 	return false;
 }
 
-function userInTest(
+type NoParticipation = { type: 'NO_PARTICIPATION' };
+
+type ActiveParticipation =
+	| { type: 'ACTIVE_PARTICIPATION' }
+	| {
+			type: 'REFERRER_CONTROLLED_ACTIVE_PARTICIPATION';
+			acquisitionDataTest: AcquisitionABTest;
+	  };
+
+type UserParticipation = NoParticipation | ActiveParticipation;
```

Comment:
I've split out the `NoParticipation` and `ActiveParticipation` so that `assignUserToVariant` can explicitly accept an `ActiveParticipation`. The main benefit though is that for a `referrerControlled` test we store the matching `acquisitionDataTest`. This means we don't have to search for it again in `assignUserToVariant` and deal with the fact that the types signature of `.find` returns something that might be `undefined`